### ISSUE TITLE
Homematic Addon - [ERROR] BidCosRpcDispatcher - Could not register BidCos-RF_java

### ISSUE THREAD

**Issue by fabsss** [home-assistant/hassio-addons#842]
I tried to install homematic-addon with "HM-MOD-RPI-PCB". I get the following log when starting the addon. I can open the ReGa interface but it fails to open the device registration page. 

My setup is hass.io docker installation on raspbian buster on a raspberry pi 3. I also added the "dtoverlay=pi3-miniuart-bt" line to the config.txt.

Log:

2019-11-19 18:55:55,191 **[ERROR] BidCosRpcDispatcher - Could not register BidCos-RF_java**
Nov 19, 2019 6:55:55 PM io.vertx.core.impl.BlockedThreadChecker
WARNING: Thread Thread[vert.x-eventloop-thread-2,5,main] has been blocked for 9288 ms, time limit is 2000
io.vertx.core.VertxException: Thread blocked
	at java.base@11.0.4/jdk.internal.misc.Unsafe.park(Native Method)
	at java.base@11.0.4/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:234)
	at java.base@11.0.4/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2123)
	at java.base@11.0.4/java.util.concurrent.LinkedBlockingQueue.poll(LinkedBlockingQueue.java:458)
	at app//de.eq3.ccu.virtualdevice.service.internal.rega.VirtualDeviceHandlerRega.registerCallback(VirtualDeviceHandlerRega.java:99)
	at java.base@11.0.4/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.4/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.4/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.4/java.lang.reflect.Method.invoke(Method.java:566)
	at app//de.eq3.cbcs.legacy.communication.rpc.internal.rpc.AnnotationAwareRpcHandler.execute(AnnotationAwareRpcHandler.java:80)
	at app//de.eq3.ccu.server.internal.RpcMessageHandler.handle(RpcMessageHandler.java:70)
	at app//de.eq3.ccu.server.internal.RpcMessageHandler.handle(RpcMessageHandler.java:24)
	at app//io.vertx.core.eventbus.impl.HandlerRegistration.deliver(HandlerRegistration.java:212)
	at app//io.vertx.core.eventbus.impl.HandlerRegistration.handle(HandlerRegistration.java:191)
	at app//io.vertx.core.eventbus.impl.EventBusImpl.lambda$deliverToHandler$3(EventBusImpl.java:505)
	at app//io.vertx.core.eventbus.impl.EventBusImpl$$Lambda$148/0x6cf48628.handle(Unknown Source)
	at app//io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:337)
	at app//io.vertx.core.impl.ContextImpl$$Lambda$81/0x6e638828.run(Unknown Source)
	at app//io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at app//io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at app//io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:445)
	at app//io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.base@11.0.4/java.lang.Thread.run(Thread.java:834)

2019-11-19 18:56:00,191 **[ERROR] BidCosRpcDispatcher - Could not register BidCos-RF_java**

**Comment by Vertux** [comment]
I can confirm this issue, it still presists - [this](https://github.com/eq-3/occu/issues/90) and [this]( #971) is related. I got the same OS + hardware combination as @fabsss

**Comment by felixsteghofer** [comment]
can also confirm. Raspi3 with pretty much default config and `dtoverlay=pi3-miniuart-bt` in /mnt/boot/config.txt. 
Tried also setting `enable_uart=1` in config.txt but no combination working.
The HM-MOD-RPI-PCB is working fine in raspimatic.

Also tried all recommendations in this [thread](https://community.home-assistant.io/t/using-bluetooth-and-dtoverlay-pi3-miniuart-bt/63971/4), still no sign of CCU2 access; bluetooth (desirable) broken with 
```
dtoverlay=pi3-miniuart-bt
enable_uart=1
```

**Comment by FloThinksPi** [comment]
Yea i also have this both on the RPI3 and RPI4 with HM-MOD-RPI-PCB. It wont recognize it however it works with pivccu3 and raspmatic.
Also tooling inside the container cannot propperly reach the module i guess
![image](https://user-images.githubusercontent.com/5863788/77568793-77772400-6ec9-11ea-9480-a673eaf335b1.png)

**Comment by felixsteghofer** [comment]
@pvizeli anything I can help to resolve this issue? I did some more basic tests but couldn't get the module to be accessible inside the addons container.

**Comment by elcajon** [comment]
I got the Homematic addon to work with HM-MOD-RPI-PCB  on RPI4 (Raspberry Pi OS (32-bit) Lite)

sudo bash -c 'cat << EOT >> /boot/config.txt
dtoverlay=pi3-disable-bt
EOT'
sudo systemctl disable hciuart.service

apt install build-essential bison flex libssl-dev -y
wget -q -O - https://www.pivccu.de/piVCCU/public.key | sudo apt-key add -
bash -c 'echo "deb https://www.pivccu.de/piVCCU stable main" > /etc/apt/sources.list.d/pivccu.list'
apt update
apt install raspberrypi-kernel-headers pivccu-modules-dkms -y
apt install pivccu-modules-raspberrypi -y 
sed -i 's/dtoverlay=pivccu-raspberrypi/#dtoverlay=pivccu-raspberrypi/' /boot/config.txt
sed -i /boot/cmdline.txt -e "s/console=serial0,[0-9]\+ //"
sed -i /boot/cmdline.txt -e "s/console=ttyAMA0,[0-9]\+ //"

**Comment by obfri** [comment]
Same problem with HMIP-RFUSB stick - don't believe that there is a power supply issue... Any suggestions what to do? Using elcajon's solution?

**Comment by elcajon** [comment]
> Same problem with HMIP-RFUSB stick - don't believe that there is a power supply issue... Any suggestions what to do? Using elcajon's solution?

Give it a try, work like a charm for me.

**Comment by FloThinksPi** [comment]
@elcajon still cannot get it to work. I started with a clean install of rasbian on pi4 and installed homeasistant via the supervisor method. When adding the plugin(according to docs), same error as usual even when installing pivccu kernel modules but not loading them as you showed in your example. Maybe its because i previously ran a newer version of ccu which updated the firmware on the HM-MOD-RPI-PCB module ? 🤔 I think i will again install pivccu3 and nginx to reverse-proxy 🤷.

**Comment by FloThinksPi** [comment]
I found out what was wrong. Apparently when using rapsmatic or pivccu the HM-MOD-RPI-PCB Module gets a new firmware that is incompatible with plain OCCU. One has to manually flash the OCCU Firmware onto the module otherwise strange issues appear like unparsable commands and so on.
I made a PR to document this and add an example instruction how to flash the module from HassOS with the help of a container. See #1726
